live-hls.js

```diff
@@ -604,7 +609,7 @@ const parseMaster = function(request, response, body) {
             line = trimCharacters(lines[i].substr(uriIndex + 5), ['\'', '/', '.']).replace(/['"]+/g, '').replace(/(\r)/gm,"");
             indexOfLastSlash = fullurl.lastIndexOf('/');
             baseurl = fullurl.slice(0, indexOfLastSlash) + '/';
-            manifestUrl = `http://${request.headers.host}/${eventType}?url=${baseurl + trimCharacters(line, ['.', '/'])}`;
+            manifestUrl = `${protocol}://${request.headers.host}/${eventType}?url=${baseurl + trimCharacters(line, ['.', '/'])}`;
```

Comment:
This `protocol` is for the endpoint where murphy server is hosted - not for the origin url of the master manifest. So, changing it to match the manifest URL is incorrect.

Murphy intercepts the master and child manifests to create rolling windows out of the VOD playlists.

Reply:
Ideally, the murphy server endpoint should be picked up from some sort of configuration - so that we don't include logic in the code to construct the base URL.
We can then include the `protocol`/`scheme` in the base url setting in this configuration.

Reply:
If murphy is hosted behind a https endpoint the existing code won't work as it always requests http. This was meant as a easy fix for this problem.